Feature/explorer frontend updates

AshTNA opened the pull request:
Hi @gtvj & @danbentley 

Here is the PR for the Explorer frontend updates in Wagtail.

I've had to tweak the SASS/HTML a bit more than expected to make everything work, as the HTML has become out of sync with `ds-wagtail-frontend` (which I'll try and resynchronise at some point).

The changes include:

- An updated header with some descriptive text. The logo has been fixed too (the "BETA" text was serif outside of Chrome)
- Updates to the `card-group-promo` component styles, to allow for the new index page card styles
- Padding fixes on the `card-group-secondary-nav` and `card-group-record-summary` cards
- Adding the amount of items returned on the results page, which also involved some styling changes to the `results_options` component. 

I've hard coded some H2 elements into a couple of the blocks/templates. They're needed for accessibility - they'll probably need to be added to the pages model.

Below I've pasted a screenshot of the `results_options` component on `results_page.html`. The model validation prevents you from creating a `results_page` currently, so I used the `urls.py` to check that page. It's also slightly different from `ds-wagtail-frontend` as in that repo I forgot to account for the non-JS submit button!

#### Results options on Desktop
![image](https://user-images.githubusercontent.com/8880610/125757959-519e8328-27a0-45d6-8002-b46c8b8b14b2.png)
#### Results options on Mobile (320px)
![image](https://user-images.githubusercontent.com/8880610/125758545-d20ab2d8-c41c-4ae9-bdd8-3f9ae02cc1dc.png)


Would you be able to approve the changes if all is OK? Thanks :+1:

gtvj commented on sass/includes/_header.scss:
```diff
@@ -1,9 +1,48 @@
 .tna-header {
+  padding: 1rem 3rem 0rem 3rem;
```
I don't think you need the `rem` unit on the `0` here. `padding: 1rem 3rem 0 3rem` should be fine.

gtvj commented on sass/includes/_header.scss:
```diff
@@ -1,9 +1,48 @@
   }
+
+  &__description {
+    font-size: 1.2em;
```
Could we change this to use `rem`

gtvj commented on sass/includes/_header.scss:
```diff
@@ -1,9 +1,48 @@
+
+    &__description {
+      padding-top: 0;
+      padding-left: 5em;
```
Could we change this to use `rem`?

gtvj commented on sass/includes/_header.scss:
```diff
@@ -1,9 +1,48 @@
+    &__description {
+      padding-top: 0;
+      padding-left: 5em;
+      max-width: 600px;
```
Could we use a `rem` value here? I think it's `37.5rem`

gtvj commented on sass/includes/_results-options.scss:
```diff
@@ -1,22 +1,43 @@
 .results-options {
-  text-align: right;
+  background-color: $color__green;
+  padding: 1em;
```
Could we use `rem` here

gtvj commented on templates/collections/blocks/featured_page.html:
```diff
@@ -1,3 +1,3 @@
-<ul class="card-group">
+<ul class="row card-group">
```
We're mixing BEM and Bootstrap here and should separate them so that the `row` is on a parent - `<div>` perhaps - and the component is within. See: https://github.com/nationalarchives/tna-frontend-design-toolkit/blob/main/GUIDELINES.md#avoid-mixing-bem-and-grid-classes-on-the-same-element

gtvj commented on templates/collections/blocks/promoted_pages.html:
```diff
@@ -3,13 +3,13 @@
 <p>{{ value.sub_heading }}</p>
 
-<ul class="row card-group">
+<ul class="row card-group--list-style-none">
```
Similar to the comment above. We should avoid mixing BEM and Bootstrap on the same element.

danbentley reviewed (approved):
Thanks Ash. Looks Great!

gtvj commented on templates/collections/blocks/time_period_explorer.html:
```diff
@@ -5,11 +5,11 @@
         <div class="row">
             <div class="col-md-12 col-lg-7 col-xl-6">
-                <a href="{{ value.page.url }}" class="card-group-promo__card-link">
+                <a href="{{ value.page.url }}" class="card-group-promo__card-link" tabindex="-1">
```
More of a comment than anything else, but we've removed the image link on the basis that the link is also presented in the text below and this avoids the user having to tab multiple times for each card.

gtvj commented on templates/collections/explorer_index_page.html:
```diff
@@ -5,13 +5,12 @@
-            </div>
-        </div>
+    <div class="container mt-4">
+       <ul class="row card-group">
```
Similar to the comments above, we should avoid mixing BEM and Bootstrap on the same element.

gtvj commented on templates/collections/results_page.html:
```diff
@@ -14,7 +14,7 @@
                     <h3 class="sr-only">Options</h3>
                     {% include 'includes/results-options.html' %}
-                    <ul class="row card-group">
+                    <ul class="row card-group--list-style-none">
```
Similar to comments above, we should avoid mixing BEM and Bootstrap on the same element.

gtvj commented on templates/collections/time_period_explorer_index_page.html:
```diff
@@ -5,16 +5,16 @@
-            <ul class="row card-group">
+    <div class="container mt-4">
+            <h2 class="sr-only">Select a time period</h2>
+            <ul class="row card-group--list-style-none">
```
Similar to comments above, we should avoid mixing BEM and Bootstrap on the same element.

gtvj commented on templates/collections/topic_explorer_index_page.html:
```diff
@@ -5,16 +5,16 @@
-        <div class="row">
-            <ul class="row card-group">
+    <div class="container mt-4">
+            <ul class="row card-group--list-style-none">
```
Similar to comments above, we should avoid mixing BEM and Bootstrap on the same element.

gtvj commented on templates/includes/card-group-promo--dark.html:
```diff
@@ -1,33 +1,32 @@
-                        <source media="(max-width: 991px)" srcset="{{ teaser_image_extra_small.url }}">
+            <div class="col-md-12 col-lg-7 col-xl-6">
+                <a href="{{ page.url }}" tabindex="-1" class="card-group-promo__card-link">
+                    <picture class="mt-auto">
```
Rather than using Bootstrap's `mt-auto` here, because this padding is a responsibility of the component I think the margin styles should be within the Sass for the component. I think it's fine to use something like `class="container mt-4"` as you have elsewhere because the relevant element is being controlled solely by Bootstrap. Hope that doesn't sound contradictory - happy to explain on a call if that would be helpful.

AshTNA replied:
Thanks @gtvj, looks like the class is not needed with the new styles anyway so I have removed it.

gtvj commented on templates/includes/card-group-promo--green.html:
```diff
@@ -0,0 +1,33 @@
+        <div class="row">
+            <div class="col-md-12 col-lg-7 col-xl-6">
+                <a href="{{ page.url }}" tabindex="-1" class="card-group-promo__card-link">
+                    <picture class="mt-auto">
```
Similar to the comment above, I think the margin for this `picture` element should be within the Sass for the component.

AshTNA replied:
Thanks @gtvj, looks like the class is not needed with the new styles anyway so I have removed it.

gtvj commented on templates/includes/related-content.html:
```diff
@@ -12,8 +12,8 @@ <h3 class="sr-only">Event - ‘Before Shakespeare’ at The National Archives</h
             <div class="card-group-secondary-nav__body">
-                <h3 class="tna-card__heading" aria-hidden="true">Event - ‘Before Shakespeare’ at The National Archives</h3>
-                <p><time datetime="2018-08-01">Wednesday 1 August 2018</time></p>
+                <h3 class="card-group-secondary-nav__heading" aria-hidden="true">Event - ‘Before Shakespeare’ at The National Archives</h3>
```
Just a quick question - why are we hiding this from assistive technology with `aria-hidden`?

AshTNA replied:
Hi @JamesWChan, would you be able to answer this?

JamesWChan replied:
I copied this code directly from card-group-secondary-nav.php, I haven't changed it. There is an`<h3>` on line 4 with a class of `sr-only` so I guess this is to prevent screen readers from reading the heading out twice. If you want it changed just let me know.

AshTNA replied:
Thanks @JamesWChan 

I'll also need to update this file so that the card is an `<li>` instead of a `<div>`, as we made all the cards sit within a `<ul>` in `ds-wagtail` but I didn't reflect this in `ds-wagtail-frontend` (apologies) - it shouldn't affect anything visually on insights as the SASS in this repo is accounting  for the `<ul>` version of the cards.

JamesWChan replied:
No worries 👍

gtvj replied:
I've been thinking about the doubling of `<h3>` elements within this component and have spoken to @matt-blair about it. Matt has said that he sees the images as being decorative, in which case we can treat them as such - see https://www.w3.org/WAI/tutorials/images/decorative/ 

If we do that we would not need the initial `<h3>` and could treat the second one as a normal heading. This seems OK to me.

@AshTNA @JamesWChan - what do you think?

AshTNA replied:
Thanks @gtvj - that makes sense to me, though I don't know much about this component in general (aside from the card stuff)

JamesWChan replied:
Sounds good to me. In which case, should we also remove the alt text as per the W3 advice?

gtvj replied:
Yes - we'd need to set `alt=""` and remove the heading. If we're agreed, I think we should proceed with this change for all cards that have this duplicate heading pattern.

gtvj reviewed (changes requested):
Hi Ash, good work on this PR. Thanks. I've requested a few changes and made a couple of observations. There is also a question or two I've asked.

gtvj commented:
@AshTNA - one other thing to bear in mind is that, for the sorting we'll need to think about how we're going to support [WCAG 2.1 Success Criterion 3.2.2 (Level A)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-unpredictable-change.html) in the JS version. While this is a pattern we've used before, the testing of KBS revealed it to be a failure. Our options are either to: 

- Warn the user that there's going to be a change of context, or;
- Not cause a change of context (i.e. asynchonously load the updated results)

Happy to discuss. 

All the best, 
Gwyn

AshTNA commented:
Hi @gtvj,

Thanks for the comments - I have changed the `em` values to `rem`, 0 values, and removed the `mt-auto` from the promo cards.

I'll look into the `<ul class="row card-group>` issue. This was introduced when we conducted the accessibility fixes for the Explorer so it was approved previously in a PR at some point.When I initially implemented that, I was aware that CB's guidance said _avoid_ instead of strictly outlawing it, which is why I believe it was merged when it was first introduced. If I recall correctly, the styles go a bit awry when the `row`  is in a parent `<div>`. And of course a `<ul>` can only have `<li>` elements as its children. I will try and fix it but it may be easier to look at this on a call together - I don't think I figured out the exact reason it was giving  weird behaviour.

Thanks :+1:

gtvj commented:
Thanks @AshTNA - on the BEM/Bootstrap stuff, sorry to ask you to fix something that had previously been merged in another PR but as I reflect on this (while we feel our way through the new approaches) I would like to keep the BEM and Bootstrap separate if we possibly can. In this instance I'd suggest introducing a containing `<div>` to hold the Bootstrap class - any issues are likely to be a margin or padding. Happy to jump on a call to look at it together if you'd like.

AshTNA commented:
Thanks @gtvj - I've got it working by:

- Separating the `row` out into a parent div
- Adding ` display: flex;` `flex-wrap: wrap;` and `width: 100%;` to the `card-group` which mimics what the `row` was doing when both classes were on the same element.

I've also checked the results cards using the Kong environment variables, and they seem to be OK as well.

gtvj commented:
All merged @AshTNA - thanks for your work on this 👍🏽
